flash.go

```diff
@@ -0,0 +1,84 @@
+package buffalo
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+// FlashPrefix is the prefix inside the Session.
+const FlashPrefix = "_flash_"
+
+//Flash is a struct that helps with the operations over flash messages.
+type Flash struct {
+	data map[string][]string
+}
+
+//Delete removes a particular key from the Flash.
+func (f *Flash) Delete(key string) {
+	delete(f.data, key)
+}
+
+//Clear removes all keys from the Flash.
+func (f *Flash) Clear() {
+	f.data = map[string][]string{}
+}
+
+//Set allows to set a list of values into a particular key.
+func (f *Flash) Set(key string, values []string) {
+	f.data[key] = values
+}
+
+//Add adds a flash value for a flash key, if the key already has values the list for that value grows.
+func (f *Flash) Add(key, value string) {
+	if len(f.data[key]) == 0 {
+		f.data[key] = []string{value}
+		return
+	}
+
+	f.data[key] = append(f.data[key], value)
+}
+
+//Persist the flash inside the session.
+func (f *Flash) Persist(session *Session) {
+	for k := range session.Session.Values {
```

Comment:
You can clean all of this up by marshaling the flash and put just one thing into the session. Perhaps using json.